src/ApplicationInsights.ServiceFabric.Native/Net45/DependencyTrackingModule/ServiceRemotingClientEventListener.cs

```diff
@@ -0,0 +1,249 @@
+﻿using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+using Microsoft.ApplicationInsights.ServiceFabric.Remoting.Activities;
+using Microsoft.ServiceFabric.Services.Remoting;
+using Microsoft.ServiceFabric.Services.Remoting.V2;
+using Microsoft.ServiceFabric.Services.Remoting.V2.Client;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace Microsoft.ApplicationInsights.ServiceFabric.Module
+{
+    internal class ServiceRemotingClientEventListener : IDisposable
+    {
+        private readonly IEnumerable<string> correlationDomainExclusionList;
+        private readonly bool setComponentCorrelationHttpHeaders;
+        private readonly ICorrelationIdLookupHelper correlationIdLookupHelper;
+        private readonly TelemetryClient client;
+        private readonly TelemetryConfiguration configuration;
+        private ConcurrentDictionary<IServiceRemotingRequestMessage, IOperationHolder<DependencyTelemetry>> pendingTelemetry = new ConcurrentDictionary<IServiceRemotingRequestMessage, IOperationHolder<DependencyTelemetry>>();
```

Comment:
It is possible (even probable as we've seen in http dependency tracking) that under some conditions stop event never happens or happens before start. 
It could happen in some corner cases because of concurrency, or just a bug in the code.

In this case pending telemetry would grow. In AI SDK they use memory cache or weak reference table to keep the telemetry. Please use one of this methods.

Reply:
well, it's not possible (even probable) that stop will come before start based on the implementation I added in SF platform. 

However, in case of failed requests - it is possible that we won't get a stop. Using MemoryCache is a good suggestion. I'll look into the AI implementation and see if I can use that.